service/src/main/java/bio/terra/workspace/service/resource/referenced/ReferencedResourceService.java

```diff
@@ -83,7 +83,11 @@ public void updateReferenceResource(
       AuthenticatedUserRequest userReq) {
     workspaceService.validateWorkspaceAndAction(
         userReq, workspaceId, SamConstants.SAM_UPDATE_REFERENCED_RESOURCE);
-    resourceDao.updateResource(workspaceId, resourceId, name, description);
+    try {
```

Comment:
Tova made a method `rethrowIfSamInterrupted` that packages this pattern up in a method that takes a function. You could add that to DbRetryUtils or elsewhere to DRY this up a bit.

Reply:
I saw that pattern and have mixed feelings. It does save a lot of repeated try/catch boilerplate, but it does also make the code a little harder for me to follow - that's a second layer of lambdas to actually execute a function. Maybe that's not a bad thing, but it does feel a little off. 
This is already what SamService does though, so I added methods to DbRetryUtils for more consistency and to cut down the repeated code.

Reply:
Yes, I share your "feels a little off", but I am slowly learning to like it. What I notice is: if we decide we want to throw a different exception, it is a change in one place.